src/ray/raylet/scheduling_policy.h

```diff
@@ -36,7 +36,20 @@ class SchedulingPolicy {
       std::unordered_map<ClientID, SchedulingResources> &cluster_resources,
       const ClientID &local_client_id);
 
-  std::vector<TaskID> SpillOver(SchedulingResources &remote_scheduling_resources) const;
+
+  /// \brief Given a set of cluster resources and subset of cluster nodes,
+  /// perform a spill-over scheduling operation. This is a light-weight decision.
+  ///
+  /// \param cluster_resources: a set of cluster resources containing resource and load
+  /// information for some subset of the cluster. For all client IDs in the returned
+  /// placement map, the corresponding SchedulingResources::resources_load_ is
+  /// incremented by the aggregate resource demand of the tasks assigned to it.
+  /// \param remote_client_ids a list of IDs corresponding to raylets that participate
+  /// in this scheduling decision.
+  /// \return Scheduling decision, mapping tasks to raylets for placement.
+  std::unordered_map<TaskID, ClientID> SpillOver(
+      std::unordered_map<ClientID, SchedulingResources> &cluster_resources,
+      std::vector<ClientID> &remote_client_ids);
```

Comment:
Is the change to consider the entire batch in `SpillOver` necessary? I prefer if we don't modify the scheduling policy at all and instead randomize the order of the batch in `HeartbeatBatchAdded`.

Reply:
I personally think the Scheduling policy should take advantage of having the entire batch to make better decisions; even though we're shuffling the vector here there could be a better solution, in which case it's better to leave it to the policy to make the decision of whether or not randomizing order is necessary. 

Randomizing it outside is possible, but to your other comment in that case we might as well just randomize at the monitor and save the nodes from the work.

Reply:
I agree, but I think we should leave it for a future PR and just try to emulate the functionality on master right now. I'm actually not so concerned about whether the randomization is expensive, just want to keep the PR as simple as possible.